code/modules/skills/skillset.dm

```diff
@@ -0,0 +1,28 @@
+/datum/skillset
+	var/list/skills
+	var/list/initial_skills
+
+/datum/skillset/New()
+	for(var/datum/skill/s as anything in initial_skills)
+		LAZYSET(skills, initial(s.name), new s)
```

Comment:
объекты скилла тоже можно синглтоном сделать так-то

Reply:
Переделал что скил тоже синглтон вот так:

```
/datum/skillset/New()
	for(var/datum/skill/s as anything in initial_skills)
		LAZYSET(skills, initial(s.name), all_skills[s])
	for(var/datum/skill/skill as anything in skills_list)
		if(!(initial(skill.name) in skills))
			LAZYSET(skills, initial(skill.name), all_skills[skill])
```
Но поскольку активные скилы мы изменяем то добавил вот это ещё в skills.dm
```
/datum/skills/New()
	for(var/skill_name in active.skills)
		var/datum/skill/original = active.get_skill(skill_name)
		var/datum/skill/skill_copy = new original.type  // new instance of skill because we will change the skills in active skillset
		LAZYSET(active.skills, skill_name, skill_copy)

```